Compile fails with openssl binary build mode

The openssl stub uses a method () that is no longer public in OpenSSL 3.0 (see https://github.com/openssl/openssl/issues/16088)

My build command:

```
(cd build \
      && cmake -DARCH=x64 -DTOOLCHAIN=GCC -DTARGET=Debug \
               -DCRYPTO=openssl -DENABLE_BINARY_BUILD=1 -DCOMPILED_LIBCRYPTO_PATH=${OPENSSL}/lib64/libcrypto.so -DCOMPILED_LIBSSL_PATH=${OPENSSL}/lib64/libssl.so \
               .. \
      && make copy_sample_key \
      && make )
```

The error:

```
[ 31%] Linking C executable ../../bin/test_spdm_requester
/tmp/test_spdm_requester.XOo3q2.ltrans0.ltrans.o: In function `libspdm_ecd_set_pub_key':
<artificial>:(.text+0xd5118): undefined reference to `evp_pkey_copy_downgraded'
/tmp/test_spdm_requester.XOo3q2.ltrans0.ltrans.o: In function `libspdm_ecd_set_pri_key':
<artificial>:(.text+0xd520a): undefined reference to `evp_pkey_copy_downgraded'
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -155,11 +155,6 @@ else()
     MESSAGE(FATAL_ERROR "Unknown CRYPTO")
 endif()
 
-if(DISABLE_EDDSA STREQUAL "1")
-    add_definitions(-DLIBSPDM_EDDSA_ED25519_SUPPORT=0)
-    add_definitions(-DLIBSPDM_EDDSA_ED448_SUPPORT=0)
-endif()
-
 if(ENABLE_BINARY_BUILD STREQUAL "1")
     if(NOT CRYPTO STREQUAL "openssl")
         MESSAGE(FATAL_ERROR "enabling binary build not supported for non-openssl")
@@ -173,11 +168,17 @@ if(ENABLE_BINARY_BUILD STREQUAL "1")
         MESSAGE(FATAL_ERROR "enabling binary build requires path to libssl.")
     endif()
 
+    # Disable EDDSA when binary builds are enabled. EDDSA may be enabled in the future.
+    add_definitions(-DLIBSPDM_EDDSA_ED25519_SUPPORT=0)
+    add_definitions(-DLIBSPDM_EDDSA_ED448_SUPPORT=0)
+
     MESSAGE("ENABLE_BINARY_BUILD=1")
     MESSAGE("COMPILED_LIBCRYPTO_PATH=${COMPILED_LIBCRYPTO_PATH}")
     MESSAGE("COMPILED_LIBSSL_PATH=${COMPILED_LIBSSL_PATH}")
+    MESSAGE("Disabling EDDSA support due to ENABLE_BINARY_BUILD=1")
 
     SET(CRYPTO_LIB_PATHS ${COMPILED_LIBCRYPTO_PATH} ${COMPILED_LIBSSL_PATH})
+
 else()
     SET(CRYPTO_LIB_PATHS ${CRYPTO}lib)
     MESSAGE("ENABLE_BINARY_BUILD=0; Building ${CRYPTO} library from source.")

